csrc/executor.cpp

```diff
@@ -257,11 +260,31 @@ void FusionExecutor::compileFusion(
   NVF_ERROR(
       !fusion->outputs().empty(), "No output found for this kernel, aborting.");
 
+  // TODO: refactor the options_ passed through
+  options_.device =
+      c10::Device(c10::DeviceType::CUDA, (int8_t)args.getDeviceIndex());
+
```

Comment:
I moved this up so `compileFusion` sets up `options_.device` correctly for `runFusion`.